moveit_ros/planning/srv_kinematics_plugin/src/srv_kinematics_plugin.cpp

```diff
@@ -71,19 +72,15 @@ bool SrvKinematicsPlugin::initialize(const std::string &robot_description,
   setValues(robot_description, group_name, base_frame, tip_frames, search_discretization);
 
   ros::NodeHandle private_handle("~");
-  rdf_loader::RDFLoader rdf_loader(robot_description_);
-  const boost::shared_ptr<srdf::Model> &srdf = rdf_loader.getSRDF();
-  const boost::shared_ptr<urdf::ModelInterface>& urdf_model = rdf_loader.getURDF();
 
-  if (!urdf_model || !srdf)
+  // Check robot_state is initialized
+  if (!robot_state_)
   {
-    ROS_ERROR_NAMED("srv","URDF and SRDF must be loaded for SRV kinematics solver to work."); // TODO: is this true?
-    return false;
+    robot_state_.reset(new robot_state::RobotState(robot_model->getConstPtr()));
```

Comment:
Does this create a `shared_ptr` loop? `RobotModel` contains `JointModelGroup`, which has the kinematics solver in the `group_kinematics_` member variable. Since this kinematics solver will now contain a `shared_ptr` to a `RobotState` which keeps a `shared_ptr` to the `RobotModel`, doesn't that create a loop?